src/app.ts

```diff
@@ -14,6 +15,14 @@ export type AppState = DefaultState;
 
 export type AppContext = RouterContext<AppState, {
   articles: Articles;
+  dataFactory: DataFactory;
+  datasetFactory: DatasetCoreFactory;
+  request: {
```

Comment:
I guess the tests have some reset mechanism for these `request` and `response`, or they build a new `AppContext` every time?

Reply:
Context is fresh for every request handled.